mentat/auto_completer.py

```diff
@@ -0,0 +1,276 @@
+import os
+import shlex
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Set, TypedDict, cast
+
+import attr
+from pygments.lexer import Lexer
+from pygments.lexers import guess_lexer_for_filename
+from pygments.token import Token
+from pygments.util import ClassNotFound
+
+from mentat.command.command import Command
+from mentat.session_context import SESSION_CONTEXT
+from mentat.utils import get_relative_path
+
+SECONDS_BETWEEN_REFRESH = 5
+
+whitespace = " \t\r\n"
```

Comment:
The reason I use this is actually because it's the same whitespace string that shlex uses; I could use .isspace but I think I'll keep this just for slight consistency with shlex.